EIPS/eip-4362.md

```diff
@@ -0,0 +1,270 @@
+      - Sort the strings in `actionSet` alphabetically.
+      - Let `actions` be the string concatenation of each action in the array with the delimiter `", "`.
+      - Let `targets` be the string concatenation of each key in `targetedActions` with the delimiter `", "`, for those keys such that the associated value (or any permutation of that value) is identical to `actionSet`.
+      - Append the string concatenation of `" ("`, `numbering`, `")"` to `capgrok-transformed-statement`.
+      - Append `namespace` concatenated with `": "` to `capgrok-transformed-statement`.
+      - Append `actions` to `capgrok-transformed-statement`.
+      - Append the string `" for "` to `capgrok-transformed-statement`.
+      - Append `targets` to `capgrok-transformed-statement`.
+      - Append the string `" ."` to `capgrok-transformed-statement`.
+      - Increase `numbering` by 1.
+- Return `capgrok-transformed-statement`.
+
+#### CapGrok Verification
+
+The following algorithm or an algorithm that produces the same output MUST be performed to verify a SIWE CapGrok.
+
+Inputs:
+- Let `capgrok-siwe` be the input SIWE message conforming to EIP-4361 and this EIP.
+Algorithm:
+- Let `uri` be the uri field of `capgrok-siwe`.
+- Let `capgrok-uris` be an array of CapGrok URIs taken in order from the resources field of `capgrok-siwe`, such that URIs which are not valid CapGrok URIs are ignored.
+- Let `capgrok-transformed-statement` be the result of performing the above `CapGrok Translation Algorithm` with `uri` and `capgrok-uris` as input.
+- Assert that the statement field of `capgrok-siwe` ends with `capgrok-transformed-statement`.
+
+### Implementer's Guide
+
+TBD
+
+#### Web3 Application Implementers
+
+TBD
+
+#### Wallet Implementers
+
+TBD
+
+#### Protocol or API Implementers
+
+TBD
+
+## Rationale
+
+While SIWE focuses on authenticating the Ethereum account against the service (relying party or SIWE client) initiating the SIWE flow, there is no canonical way to interact with a third-party service (resource service) on behalf of the authenticated Ethereum account. For example, a relying party might want to interact with another service on behalf of the Ethereum account, for example a service that provides data storage for the Ethereum account. This specification introduces a mechanism, that allows the service (or more generally a delegee) to combine authentication and authorization of such while preserving security and optimizing UX.
+
+Note, this approach is a similar mechanism to combining OpenID Connect (SIWE auth) and OAuth2 (SIWE CapGrok) whereas SIWE CapGrok follows an Object Capability-based approach.
+
+
+## Security Considerations
+
+Resource service implementer's should not consider CapGroks as bearer tokens but instead require to authenticate the delegee in addition. The process of authenticating the delegee against the resource service is out of scope of this specification and can be done in various different ways.
```

Comment:
As someone not familiar with authentication systems, I find the first sentence unnerving. Would you mind expanding it a bit, and perhaps explain mitigations or provide examples of what "require to authenticate the delegee in addition" means?